lib/archethic_web/graphql_schema/resolver.ex

```diff
@@ -212,6 +214,37 @@ defmodule ArchethicWeb.GraphQLSchema.Resolver do
     }
   end
 
+  def beacon_chain_summary(datetime) do
+    current_datetime = DateTime.utc_now()
+
+    next_datetime_summary_time =
+      datetime
+      |> BeaconChain.next_summary_date()
+
+    previous_current_date_summary_time =
+      current_datetime
+      |> BeaconChain.next_summary_date()
+
+    authorized_nodes = P2P.authorized_and_available_nodes()
+
+    cond do
```

Comment:
Ithink it would be better to use a `case` statement instead of a `cond`, so the `DateTime.compare` will be done only once and not for each cond if the previous is false

Reply:
yep I had different conditions in version 1, will fix it.